src/jshint.js

```diff
@@ -4242,7 +4242,18 @@ var JSHINT = (function() {
       // ImportClause :: ImportedDefaultBinding
       this.name = identifier();
       addlabel(this.name, { type: "unused", token: state.tokens.curr });
-    } else if (state.tokens.next.id === "*") {
+      if (state.tokens.next.value === ",") {
+        // ImportClause :: ImportedDefaultBinding , NameSpaceImport
```

Comment:
Another way to say this is that these are equivalent:

```js
import fooDefault, * as fooAll from "foo"
// vs
import * as fooAll from "foo"
const fooDefault = fooAll.default;
```

As are these:

```js
import fooDefault, { x as fooX } from "foo"
// vs
import { default as fooDefault, x as fooX } from "foo"
```

The grammar just provides a shorthand for these cases.